OpenRA.Mods.Common/Widgets/ViewportControllerWidget.cs

```diff
@@ -315,7 +315,10 @@ public override bool HandleMouseInput(MouseInput mi)
 		{
 			if (mi.Event == MouseInputEvent.Scroll && mi.Modifiers.HasModifier(Game.Settings.Game.ZoomModifier))
 			{
+				var oldMousePosition = worldRenderer.Viewport.ViewToWorldPx(mi.Location);
 				worldRenderer.Viewport.AdjustZoom(mi.Delta.Y * Game.Settings.Game.ZoomSpeed);
+				var newMousePosition = worldRenderer.Viewport.ViewToWorldPx(mi.Location);
+				worldRenderer.Viewport.Scroll(oldMousePosition - newMousePosition, true);
```

Comment:
Alright, the problem here is that `Scroll` divides the delta by the zoom level. This can be easily fixed by multiplying by the zoom level here, which results in the correct behaviour. However, this makes me think that maybe we shouldn't be using the `Scroll` method for this in the first place.

Reply:
May be better to add a `center` argument to `AdjustZoom` and let that handle it internally?